src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -148,18 +151,24 @@ trait WorkflowComponent {
       workflowMessageQuery ++= messages.map { message => WorkflowMessageRecord(workflowId, message.value) }
     }
 
-    def update(workspaceContext: SlickWorkspaceContext, submissionId: UUID, workflow: Workflow): ReadWriteAction[Workflow] = {
-      uniqueResult[WorkflowRecord](findWorkflowByEntity(submissionId, workspaceContext.workspaceId, workflow.workflowEntity.get.entityType, workflow.workflowEntity.get.entityName)) flatMap {
-        case None => throw new RawlsException(s"workflow does not exist: ${workflow}")
-        case Some(rec) =>
-          deleteWorkflowAttributes(rec.id) andThen
-          deleteMessagesAndInputs(rec.id) andThen
-          saveInputResolutions(workspaceContext, workflow.inputResolutions, Left(WorkflowId(rec.id))) andThen
-          saveMessages(workflow.messages, rec.id) andThen
-          findWorkflowById(rec.id).map(u => (u.status, u.statusLastChangedDate)).update((workflow.status.toString, new Timestamp(workflow.statusLastChangedDate.toDate.getTime)))
+    def updateStatus(workflow: WorkflowRecord, newStatus: WorkflowStatus): ReadWriteAction[Int] = {
+      sqlu"update WORKFLOW set status = ${newStatus.toString}, status_last_changed = ${new Timestamp(System.currentTimeMillis())}, record_version = record_version + 1 where id = ${workflow.id} and record_version = ${workflow.recordVersion}" map {
+        case 0 => throw new RawlsConcurrentModificationException(s"could not update $workflow because its record version has changed")
+        case success => success
       }
-    } map { _ => workflow }
+    }
 
+    //input: old workflow records, and the status that we want to apply to all of them
+    def batchUpdateStatus(workflows: Seq[WorkflowRecord], newStatus: WorkflowStatus): ReadWriteAction[Int] = {
+      val baseUpdate = sql"update WORKFLOW set status = ${newStatus.toString}, status_last_changed = ${new Timestamp(System.currentTimeMillis())}, record_version = record_version + 1 where (id, record_version) in ("
+      val workflowTuples = workflows.map { case wf => sql"(${wf.id}, ${wf.recordVersion})" }.reduce((a, b) => concatSqlActionsWithDelim(a, b, sql","))
+      concatSqlActions(concatSqlActions(baseUpdate, workflowTuples), sql")").as[Int] flatMap { rows =>
+        if(rows.sum == workflows.size)
```

Comment:
you're right. I thought the raw sql was returning something like Vector(1,1,1,1,1) but it's actually returning Vector(5). Makes sense because it's a batch